app/models/child_object.rb

```diff
@@ -96,9 +96,21 @@ def copy_to_access_master_pairtree
   # rubocop:enable  Metrics/MethodLength
   # rubocop:enable  Layout/LineLength
 
-  def access_master_checksum_matches?
-    access_master_checksum = Digest::SHA1.file(access_master_path).to_s
-    checksum == access_master_checksum
+  def checksum_matches?
```

Comment:
Since calculating the checksum is pretty expensive, you should calculate them only if they are needed in the if statements.